server/src/test/java/org/elasticsearch/cluster/service/MasterServiceTests.java

```diff
@@ -2121,6 +2121,56 @@ public void onFailure(Exception e) {
         }
     }
 
+    public void testTimeoutRejectionBehaviourAtSubmission() {
+
+        final var actionCount = new AtomicInteger();
+
+        final var deterministicTaskQueue = new DeterministicTaskQueue();
+        final var threadPool = deterministicTaskQueue.getThreadPool(r -> {
+            if (r.toString().startsWith("master service timeout handler for [batched][test task]")) {
```

Comment:
Not in this patch, but should we have something like `String AbstractRunnable#identifyTask()` that returns unique(-ish) identifiers for a `Runnable`, so we don't have to do things like `toString` matching?

Reply:
That's kinda what `toString()` is for. It's just that the automatically generated unique identifier is kinda useless.

Reply:
I was thinking something that didn't involve string parsing -- toString seems geared towards log messages, not equality checks.

Reply:
I know what you mean, I would do something more robust if this were production code for sure. On balance I think this is best for testing purposes tho - the alternative adds a little too much stuff to production just to support this test.